app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/AbstractChartFragment.java

```diff
@@ -512,7 +510,7 @@ protected void refresh() {
                         }
                         activityEntries.add(createLineEntry(value, ts));
                 }
-                if (hr && sample.getKind() != ActivityKind.TYPE_NOT_WORN && HeartRateUtils.isValidHeartRateValue(sample.getHeartRate())) {
+                if (hr && sample.getKind() != ActivityKind.TYPE_NOT_WORN && HeartRateUtils.getInstance().isValidHeartRateValue(sample.getHeartRate())) {
```

Comment:
Not sure if the getInstance() call would be optimized by the runtime or art. We could remember the instance outside the loop.

Reply:
yes this should be cleaner this way. Will rework this.